kroxylicious-authorizer/src/main/java/io/kroxylicious/authorizer/service/ResourceType.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.authorizer.service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * An Operation is an {@code enum} of the possible operations on a resource of a particular type.
+ * We use a one-enum-per-resource-type pattern so that the {@link Class} of an operation identifies the resource type.
+ * @param <S>
+ */
+public interface ResourceType<S extends Enum<S> & ResourceType<S>> {
+
+    default List<Action> actionsOf(Stream<String> names) {
+        return names.map(topicName -> new Action(this, topicName)).toList();
+    }
```

Comment:
I'm not seeing usage of this method in the PR so I'm not clear what your intended usage is but given this is binding a stream of names to an instance of an operation I wonder if 

```suggestion
    default List<Action> actionsOn(Stream<String> names) {
        return names.map(topicName -> new Action(this, topicName)).toList();
    }
```
or possibly
```suggestion
    default List<Action> operationsOn(Stream<String> names) {
        return names.map(topicName -> new Action(this, topicName)).toList();
    }
```

might be more intention revealing names or at least read more easily at call sites. 
e.g.
```java
    enum KafkaResources implements ResourceType<KafkaResources> {
        TOPIC,
        GROUP,
    }
    
    KafkaResource.TOPIC.operationsOn(Stream.of("Topic-1", "Topic-2", .... "Topic-N"))
```